meteor/client/ui/Settings/ShowStyle/VariantSettings.tsx

```diff
@@ -169,26 +420,90 @@ export const ShowStyleVariantsSettings = withTranslation()(
 								</td>
 							</tr>
 						)}
-					</React.Fragment>
-				)
+					</tbody>
+				</React.Fragment>
+			)
+		}
+
+		VariantList = ({ children, className }) => {
+			const [, drop] = useDrop({
+				accept: ShowStyleDragDropTypes.VARIANT,
+				drop: () => this.persistStateVariants(),
+				collect: (monitor: DropTargetMonitor) => ({
+					isOver: monitor.isOver(),
+					canDrop: monitor.canDrop(),
+				}),
 			})
+
+			return (
+				<table ref={drop} className={className}>
+					{children}
+				</table>
+			)
 		}
 
-		render() {
+		VariantContainer = () => {
+			const moveVariantHandler = useCallback((dragIndex: number, hoverIndex: number) => {
+				const prevState = this.state.dndVariants
+				this.setState({
+					dndVariants: update(prevState, {
+						$splice: [
+							[dragIndex, 1],
+							[hoverIndex, 0, prevState[dragIndex] as ShowStyleVariant],
+						],
+					}),
+				})
+			}, [])
+
+			const returnVariantsForList = () => {
+				return this.state.dndVariants.map((variant: ShowStyleVariant, index: number) => (
+					<this.VariantItem
+						key={unprotectString(variant._id)}
+						index={index}
+						showStyleVariant={variant}
+						moveVariantHandler={moveVariantHandler}
+					></this.VariantItem>
+				))
+			}
+
 			const { t } = this.props
 			return (
 				<div>
-					<h2 className="mhn">{t('Variants')}</h2>
-					<table className="table expando settings-studio-showStyleVariants-table">
-						<tbody>{this.renderShowStyleVariants()}</tbody>
-					</table>
-					<div className="mod mhs">
-						<button className="btn btn-primary" onClick={this.onAddShowStyleVariant}>
-							<FontAwesomeIcon icon={faPlus} />
-						</button>
-					</div>
+					<h2 className="mhn">{t('Show Style Variants')}</h2>
+					<DndProvider backend={HTML5Backend}>
```

Comment:
This `DndProvider` is unnecessary and could cause issues. One is already registered at the root level of the application.